### DISCUSSION TITLE
Couldn't communicate with a helper application

### DISCUSSION THREAD

**Discussion by marcusheBG** [IBM/mac-ibm-notifications#172] [General]
**Describe the bug**
Receiving `"Notification Agent rich notification authorization request ended with error: Couldn't communicate with a helper application"` on recently deployed machines with signed custom complied app (not notarized)

Mosyle MDM

Experienced on a 12.6.1 and 13.2.1 Mac. ~95% of other Macs in environment working appropriately.

Tried reapplying notifications profile with reinstall profile button in MDM, no luck

Tried `sudo -u $currentUser`, no luck

Tried `launchctl asuser $currentUserUID`, worked!

After running once with `launchctl asuser`, reverting script to running notifier without arguments was working properly and providing notification.

Most Macs in MDM working, some need a "jumpstart" with `launchctl asuser`. For now, leaving `launchctl asuser` in our scripts.....

**Desktop (please complete the following information):**
 - OS: [12.6.1 and 13.2.1]
 - Project version [Version 2.9.1 Build 96]

**Additional context**
Running from Mosyle MDM. Most machines work without `launchctl asuser.` Some need a jumpstart. Just leaving `launchctl asuser` in our scripts for now so we may not see the bug again to report more.

Thank you!

**Comment by SMartorelli** [comment]
As we're unable to reproduce the issue, I'm transitioning this to a discussion thread where individuals can share their experiences. Additionally, I've established a [new wiki page](https://github.com/IBM/mac-ibm-notifications/wiki/Known-Issues-And-Fixes#2-notification-agent-rich-notification-authorization-request-ended-with-error-couldnt-communicate-with-a-helper-application---discussion) dedicated to tracking these sporadic issues, which are likely resultant from macOS anomalies or incorrect workflows.
